rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,143 @@
+
+### Minaform Restart
+
+The restart command runs a `minaform stop` followed by `minaform start` and accepts the same flags as both start + stop.
+
+### Minaform Upgrade
+
+The upgrade command runs a terraform apply, to update the existing deployment with new changes. Ideally this should also provide more debugging flags to terraform and other tooling, and require user input to proceed so that the engineer deploying can understand what state will be lost (if any) in the process. Accepts the same flags as start/stop/restart.
+
+### Minaform connect
+
+To facilitate easy local testing, `minaform connect <testnet_name>` will use the terraform.env to locally start a docker container node that connects to the deployed testnet. With some additional flags to specify keypairs to mount `--key <key folder path>` and where to persist the .coda-config dir (default nowhere) `--local-config-dir <path>`. This is another area where many developers know some process for assembling the correct docker image, daemon.json, and peers for a given network but with the terraform.env and kubectl its very possible to script and lower the barrier to entry even more.
+
+### Minaform Watch
+
+Addressing Pain Point #2 is the `minaform watch` command that will integrate with any scripts or tools we have right now (and write going forward) for monitoring network health. For grafana and google cloud it will just print links to their web dashboards. For kubernetes startup/liveness/readiness probes and tracking pod health there will be live statistics printed to the console. When new tooling is created for monitoring network health it should be added at least as a flag / subcommand of watch to allow for easy visibility.
+
+An issue has been raised that the logic for monitoring/validating the health of a network is orthogonal to deployment, which is understandable. The longer term plan of eventually transitioning to go makes the possible value of this sort of command much higher, and as such a bash implementation is less worthwhile. The grafana dashboard + gcp UI url for a given testnet can just as easily be returned by the start/restart/upgrade commands.
+
+## Beyond the minaform tool
+
+### Minaform Docker image
+
+To address pain points #7 and #8, and to some extent the larger problem of how to make it easy for anyone, from anywhere, to deploy networks, we need a minaform docker image that can be bundled alongside all of its dependencies (nodejs for coda-network, some of our deb packages like generate-keypair, terraform, helm, most of mina-automation repo, and the helm charts themselves). These dependencies can then be handled by updates to the docker image, and anyone having trouble with version mismatches can mount their credentials into a docker container and get moving quickly.
```

Comment:
+1 Wonders where or if the toolchain image fits in here though?

Reply:
The toolchain image needs all of the compile-time requirements for mina itself as we use it in CI, but I don't feel like those belong in a deployment tooling container.